server/src/main/java/org/opensearch/search/backpressure/SearchBackpressureService.java

```diff
@@ -68,9 +72,12 @@ public class SearchBackpressureService extends AbstractLifecycleComponent
     private final AtomicReference<TokenBucket> taskCancellationRateLimiter = new AtomicReference<>();
     private final AtomicReference<TokenBucket> taskCancellationRatioLimiter = new AtomicReference<>();
 
-    // Currently, only the state of SearchShardTask is being tracked.
-    // This can be generalized to Map<TaskType, SearchBackpressureState> once we start supporting cancellation of SearchTasks as well.
-    private final SearchBackpressureState state = new SearchBackpressureState();
+    private final Map<Class<? extends Task>, SearchBackpressureState> searchBackpressureStates = new HashMap<>() {
```

Comment:
Please introduce the generic interface, fe `SearchBackpressureTask`, so the remove `if instanceof`  logic everywhere